ludwig/trainers/trainer_lightgbm.py

```diff
@@ -719,17 +722,17 @@ def set_steps_to_1_or_quit(self, signum, frame):
     def _construct_lgb_params(self) -> Tuple[dict, dict]:
         output_params = {}
         feature = next(iter(self.model.output_features.values()))
-        if feature.type() == CATEGORY:
+        if feature.type() == BINARY or (hasattr(feature, "num_classes") and feature.num_classes == 2):
```

Comment:
Why do we need to check the num_classes stuff here when the feature is binary?

Reply:
a user can specify a category variable with only two classes, this is to catch that case and explicitly use the binary objective